packages/app/src/connecting/components/connecting.component.tsx

```diff
@@ -108,17 +117,30 @@ const Connecting: FunctionComponent<{
     history.push(URL_MAIN.news)
   }
 
+  const onRetry = () => {
+    // TODO: do some logic to retry connection
+    history.push(URL_ONBOARDING.connecting)
+  }
+
   return (
     <>
-      {error && <ErrorConnectingModal open closeModal={close} />}
+      {error &&
+        (syncInitialized ? (
```

Comment:
when the sync error is thrown then the `syncInitialized` cannot be true, so this condition doesn't run ever.
```suggestion
        (syncState === SynchronizationState.Error ? (
```

Reply:
But I changed the condition of setting error and now it doesn't depend on _syncInitialized_
`useEffect(() => {
    if (unlocked && syncState === SynchronizationState.Error) {
      setError(true)
    }
  }, [syncInitialized, syncState, unlocked])`

Reply:
I know, but even then 😛 because when the sync error is thrown, the reducer no set the `syncInitialized` to true 🙃

Reply:
If I change the condition as you suggested then `ErrorConnectingModal` will be never shown because `syncState === SynchronizationState.Error`  is also a condition for error state

Reply:
Added error types for better conditioning in [b5ac601](https://github.com/mudita/mudita-center/pull/638/commits/b5ac601229195b7cd4decca8e7b1a48816b6b95c)